From: Mike Nolan <mpnolan@truefitness.com>
Date: Fri, 17 Apr 2015 09:38:07 -0500
Subject: Add back delete announcement support in SAP module

---
 modules/services_discovery/sap.c |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/modules/services_discovery/sap.c b/modules/services_discovery/sap.c
index 12705d2..1e3dd4e 100644
--- a/modules/services_discovery/sap.c
+++ b/modules/services_discovery/sap.c
@@ -800,11 +800,17 @@ static int ParseSAP( services_discovery_t *p_sd, const uint8_t *buf,
              * Be used to highjack an announcement by a third party.
              * Instead we cleverly implement Implicit Announcement removal.
              *
-             * if( b_need_delete )
-             *    RemoveAnnounce( p_sd, p_sd->p_sys->pp_announces[i]);
+             if( b_need_delete )
+               RemoveAnnounce( p_sd, p_sd->p_sys->pp_announces[i]);
              * else
              */
 
+            // mpn: add back delete announcement support
+            if( b_need_delete ) {
+                RemoveAnnounce( p_sd, p_sd->p_sys->pp_announces[i]);
+                msg_Dbg( p_sd, "mpn:Got delete announcement" );
+            }
+
             if( !b_need_delete )
             {
                 /* No need to go after six, as we start to trust the
-- 
