From: Mike Nolan <mpnolan@truefitness.com>
Date: Sun, 28 Feb 2016 22:43:54 -0600
Subject: Fix for upstream bug #11854

---
 lib/video.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/lib/video.c b/lib/video.c
index 4bebce5..c358d0a 100644
--- a/lib/video.c
+++ b/lib/video.c
@@ -471,13 +471,21 @@ void libvlc_video_set_teletext( libvlc_media_player_t *p_mi, int i_page )
         return;
     }
 
-    telx = var_GetInteger( p_input_thread, "teletext-es" );
+    telx = var_GetInteger( p_input_thread, "spu-es" );
+    if (telx < 0) {
+        goto end;
+    }
     if( input_GetEsObjects( p_input_thread, telx, &p_zvbi, NULL, NULL )
         == VLC_SUCCESS )
     {
-        var_SetInteger( p_zvbi, "vbi-page", i_page );
-        vlc_object_release( p_zvbi );
+        if (p_zvbi) {
+          var_SetInteger( p_zvbi, "vbi-page", i_page );
+          vlc_object_release( p_zvbi );
+        } else {
+          msg_Dbg(p_input_thread, "Skipping libvlc_video_set_teletext because decoder object is not available\n");
+        }
     }
+end:
     vlc_object_release( p_input_thread );
 }
 
